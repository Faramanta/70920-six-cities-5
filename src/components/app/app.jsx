import {BrowserRouter, Switch, Route, Link} from "react-router-dom";
import Main from "../main/main";
import SignIn from "../sign-in/sign-in";
import Favorites from "../favorites/favorites";
import Room from "../room/room";
import {ReviewsPropTypes} from "Props";

const App = (props) => {
  const {reviews} = props;

  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/">
          <Main />
        </Route>
        <Route exact path="/login">
          <SignIn />
        </Route>
        <Route exact path="/favorites">
          <Favorites />
        </Route>
        <Route exact path="/offer/:id">
          <Room reviews={reviews} />
        </Route>
        <Route
          render={() => (
            <>
              <h1 style={{display: `block`, textAlign: `center`}}>Page not found</h1>
              <Link to="/" style={{display: `block`, textAlign: `center`, marginTop: `30px`}}>Go to home</Link>
            </>
          )}
        />
      </Switch>
    </BrowserRouter>
  );
};

App.propTypes = {
  reviews: PropTypes.arrayOf(ReviewsPropTypes).isRequired,
};

export default App;
