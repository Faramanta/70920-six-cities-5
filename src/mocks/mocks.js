export const offers = [{
  bedroomCount: 3,
  cityLocation: {latitude: 52.37454, longitude: 4.897976, zoom: 13},
  city: `Amsterdam`,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  facilities: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  ownerName: `Angelina`,
  ownerAvatar: `img/1.png`,
  isSuper: true,
  id: 1,
  images: [`img/1.png`, `img/2.png`],
  isFavorite: false,
  isPremium: false,
  coordinates: [52.35514938496378, 4.673877537499948],
  zoom: 8,
  guestCount: 4,
  previewImage: `img/1.png`,
  price: 120,
  rating: 4.8,
  title: `Beautiful & luxurious studio at great location`,
  type: `apartment`
},
{
  bedroomCount: 3,
  cityLocation: {latitude: 52.37454, longitude: 4.897976, zoom: 13},
  city: `Paris`,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  facilities: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  ownerName: `Angelina`,
  ownerAvatar: `img/1.png`,
  isSuper: true,
  id: 2,
  images: [`img/1.png`, `img/2.png`],
  isFavorite: false,
  isPremium: false,
  coordinates: [52.35514938496378, 4.673877537499948],
  zoom: 8,
  guestCount: 4,
  previewImage: `img/1.png`,
  price: 120,
  rating: 4.8,
  title: `Beautiful & luxurious studio at great location`,
  type: `apartment`
}];

export const offer = {
  bedroomCount: 3,
  cityLocation: [52.370216, 4.895168],
  city: `Amsterdam`,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  facilities: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  ownerName: `Angelina`,
  ownerAvatar: `img/1.png`,
  isSuper: true,
  id: 1,
  images: [`img/1.png`, `img/2.png`, `img/2.png`, `img/2.png`, `img/2.png`, `img/2.png`, `img/2.png`, `img/2.png`, `img/2.png`],
  isFavorite: false,
  isPremium: false,
  coordinates: [52.35514938496378, 4.673877537499948],
  zoom: 8,
  guestCount: 4,
  previewImage: `img/1.png`,
  price: 120,
  rating: 4.8,
  title: `Beautiful & luxurious studio at great location`,
  type: `apartment`
};

export const comments = [
  {
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: new Date(`Tue Oct 27 2020 12:29:21 GMT+0300`),
    id: 1,
    rating: 4,
    avatarUrl: `img/1.png`,
    userId: 4,
    isPro: false,
    name: `Max`
  }
];

export const comment = {
  comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river.`,
  date: new Date(`Thu Nov 12 2020 21:05:12 GMT+0500`),
  id: 1,
  rating: 4,
  avatarUrl: `img/1.png`,
  userId: 4,
  isPro: false,
  name: `Max`
};


export const userData = {
  avatarUrl: `img/1.png`,
  email: `Oliver.conner@gmail.com`,
  id: 1,
  isPro: false,
  name: `Oliver.conner`
};

export const favoriteOffers = [{
  bedroomCount: 3,
  cityLocation: [52.370216, 4.895168],
  city: `Amsterdam`,
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  facilities: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  ownerName: `Angelina`,
  ownerAvatar: `img/1.png`,
  isSuper: true,
  id: 1,
  images: [`img/1.png`, `img/2.png`],
  isFavorite: false,
  isPremium: false,
  coordinates: [52.35514938496378, 4.673877537499948],
  zoom: 8,
  guestCount: 4,
  previewImage: `img/1.png`,
  price: 120,
  rating: 4.8,
  title: `Beautiful & luxurious studio at great location`,
  type: `apartment`
}];
