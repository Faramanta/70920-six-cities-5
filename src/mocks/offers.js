export default [
  {
    id: 0,
    city: `Paris`,
    title: `Beautiful & luxurious studio at great location`,
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    isPremium: true,
    isFavorite: false,
    type: `Apartment`,
    rating: 5,
    bedroomCount: 3,
    guestCount: 4,
    price: 120,
    pictures: [
      `https://picsum.photos/seed/1/260/200`,
      `https://picsum.photos/seed/2/260/200`,
      `https://picsum.photos/seed/3/260/200`,
      `https://picsum.photos/seed/4/260/200`,
      `https://picsum.photos/seed/5/260/200`,
      `https://picsum.photos/seed/6/260/200`,
    ],
    facilities: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    ownerName: `Angelina`,
    ownerAvatar: `https://picsum.photos/seed/1/74/74`,
    isSuper: true,
    coordinates: [52.3909553943508, 4.85309666406198]
  }, {
    id: 1,
    city: `Cologne`,
    title: `Wood and stone place`,
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    isPremium: false,
    isFavorite: true,
    type: `Room`,
    rating: 4,
    bedroomCount: 2,
    guestCount: 3,
    price: 80,
    pictures: [
      `https://picsum.photos/seed/8/260/200`,
      `https://picsum.photos/seed/9/260/200`,
      `https://picsum.photos/seed/10/260/200`,
      `https://picsum.photos/seed/11/260/200`,
    ],
    facilities: [
      `Wi-Fi`,
      `Washing machine`,
      `Heating`,
      `Cabel TV`,
      `Fridge`,
    ],
    ownerName: `Mary`,
    ownerAvatar: `https://picsum.photos/seed/2/74/74`,
    isSuper: false,
    coordinates: [52.369553943508, 4.85309666406198]
  }, {
    id: 2,
    city: `Brussels`,
    title: `Canal View Prinsengracht`,
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    isPremium: false,
    isFavorite: true,
    type: `House`,
    rating: 4,
    bedroomCount: 2,
    guestCount: 3,
    price: 132,
    pictures: [
      `https://picsum.photos/seed/12/260/200`,
      `https://picsum.photos/seed/13/260/200`,
      `https://picsum.photos/seed/14/260/200`,
      `https://picsum.photos/seed/15/260/200`,
      `https://picsum.photos/seed/16/260/200`,
    ],
    facilities: [
      `Wi-Fi`,
      `Coffee machine`,
      `Cabel TV`,
      `Fridge`,
    ],
    ownerName: `John`,
    ownerAvatar: `https://picsum.photos/seed/3/74/74`,
    isSuper: true,
    coordinates: [52.3909553943508, 4.929309666406198]
  }, {
    id: 3,
    city: `Amsterdam`,
    title: `Nice, cozy, warm big bed apartment`,
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    isPremium: true,
    isFavorite: true,
    type: `Hotel`,
    rating: 1,
    bedroomCount: 5,
    guestCount: 2,
    price: 180,
    pictures: [
      `https://picsum.photos/seed/17/260/200`,
      `https://picsum.photos/seed/18/260/200`,
      `https://picsum.photos/seed/19/260/200`,
      `https://picsum.photos/seed/20/260/200`,
      `https://picsum.photos/seed/21/260/200`,
    ],
    facilities: [
      `Wi-Fi`,
      `Coffee machine`,
      `Cabel TV`,
      `Fridge`,
    ],
    ownerName: `Irina`,
    ownerAvatar: `https://picsum.photos/seed/4/74/74`,
    isSuper: true,
    coordinates: [52.3809553943508, 4.939309666406198]
  }, {
    id: 4,
    city: `Hamburg`,
    title: `Wood and stone place`,
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    isPremium: false,
    isFavorite: false,
    type: `Room`,
    rating: 1,
    bedroomCount: 5,
    guestCount: 2,
    price: 180,
    pictures: [
      `https://picsum.photos/seed/22/260/200`,
      `https://picsum.photos/seed/23/260/200`,
    ],
    facilities: [
      `Wi-Fi`,
      `Coffee machine`,
      `Fridge`,
    ],
    ownerName: `Oleg`,
    ownerAvatar: `https://picsum.photos/seed/5/74/74`,
    isSuper: true,
    coordinates: [52.3809553943508, 4.949309666406198]
  }, {
    id: 5,
    city: `Dusseldorf`,
    title: `Nice, cozy, warm big bed apartment`,
    description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    isPremium: false,
    isFavorite: false,
    type: `Hotel`,
    rating: 1,
    bedroomCount: 5,
    guestCount: 2,
    price: 180,
    pictures: [
      `https://picsum.photos/seed/24/260/200`,
      `https://picsum.photos/seed/25/260/200`,
      `https://picsum.photos/seed/26/260/200`,
      `https://picsum.photos/seed/27/260/200`,
      `https://picsum.photos/seed/28/260/200`,
    ],
    facilities: [
      `Wi-Fi`,
      `Coffee machine`,
      `Cabel TV`,
      `Fridge`,
    ],
    ownerName: `Irina`,
    ownerAvatar: `https://picsum.photos/seed/6/74/74`,
    isSuper: true,
    coordinates: [52.3709553943508, 4.939309666406198]
  }
];
