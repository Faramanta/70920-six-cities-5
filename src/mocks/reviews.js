export default [
  {
    id: 0,
    offerId: 0,
    authorName: `Max`,
    authorAvatar: `https://picsum.photos/seed/1/54/54`,
    rating: 4,
    date: new Date(),
    text: `Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?`
  }, {
    id: 1,
    offerId: 0,
    authorName: `Nick`,
    authorAvatar: `https://picsum.photos/seed/11/54/54`,
    rating: 1,
    date: new Date(),
    text: `Et harum quidem rerum facilis est et expedita distinctio. Nam libero tempore, cum soluta nobis est eligendi optio cumque nihil impedit quo minus id quod maxime placeat facere possimus, omnis voluptas assumenda est, omnis dolor repellendus. Temporibus autem quibusdam et aut officiis debitis aut rerum necessitatibus saepe eveniet ut et voluptates repudiandae sint et molestiae non recusandae. Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repella.`
  }, {
    id: 2,
    offerId: 0,
    authorName: `Nelli`,
    authorAvatar: `https://picsum.photos/seed/12/54/54`,
    rating: 2,
    date: new Date(),
    text: `On the other hand, we denounce with righteous indignation and dislike men who are so beguiled and demoralized by the charms of pleasure of the moment, so blinded by desire, that they cannot foresee the pain and trouble that are bound to ensue; and equal blame belongs.`
  }, {
    id: 3,
    offerId: 2,
    authorName: `Ivan`,
    authorAvatar: `https://picsum.photos/seed/13/54/54`,
    rating: 5,
    date: new Date(),
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
  }, {
    id: 4,
    offerId: 2,
    authorName: `Una`,
    authorAvatar: `https://picsum.photos/seed/14/54/54`,
    rating: 5,
    date: new Date(),
    text: `At vero eos et accusamus et iusto odio dignissimos ducimus qui blanditiis praesentium voluptatum deleniti atque corrupti quos dolores et quas molestias excepturi sint occaecati cupiditate non provident, similique sunt in culpa qui officia deserunt mollitia animi, id est laborum .`
  }, {
    id: 5,
    offerId: 2,
    authorName: `Don`,
    authorAvatar: `https://picsum.photos/seed/15/54/54`,
    rating: 2,
    date: new Date(),
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
  }, {
    id: 6,
    offerId: 3,
    authorName: `Una`,
    authorAvatar: `https://picsum.photos/seed/16/54/54`,
    rating: 5,
    date: new Date(),
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
  }, {
    id: 7,
    offerId: 3,
    authorName: `Don`,
    authorAvatar: `https://picsum.photos/seed/17/54/54`,
    rating: 5,
    date: new Date(),
    text: `But I must explain to you how all this mistaken idea of denouncing pleasure and praising pain was born and I will give you a complete account of the system, and expound the actual teachings of the great explorer of the truth, the master-builder of human happiness. No one rejects, dislikes, or avoids pleasure itself, because it is pleasure, but because those who do not know how to pursue pleasure rationally.`
  }, {
    id: 8,
    offerId: 4,
    authorName: `Una`,
    authorAvatar: `https://picsum.photos/seed/18/54/54`,
    rating: 5,
    date: new Date(),
    text: `Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem.`
  }, {
    id: 9,
    offerId: 5,
    authorName: `Una`,
    authorAvatar: `https://picsum.photos/seed/19/54/54`,
    rating: 5,
    date: new Date(),
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
  }, {
    id: 10,
    offerId: 5,
    authorName: `Don`,
    authorAvatar: `https://picsum.photos/seed/20/54/54`,
    rating: 5,
    date: new Date(),
    text: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`
  }
];
