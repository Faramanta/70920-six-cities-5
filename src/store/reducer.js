import {extend} from "../utils/utils";
import {ActionType} from "./action";
import offers from "../mocks/offers";
import {CITIES} from "../const";

const cities = CITIES;
const defaultCity = CITIES[0];
const defaultOffers = offers.filter((offer) => offer.city === CITIES[0]);

const initialState = {
  activeCity: defaultCity,
  offers: defaultOffers,
  allOffers: offers,
  cities
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_CITY:
      return extend(state, {
        activeCity: action.payload
      });
    case ActionType.GET_OFFERS_LIST:
      return extend(state, {
        offers: offers.filter((offer) => offer.city === state.activeCity)
      });
  }

  return state;
};

export {reducer};
