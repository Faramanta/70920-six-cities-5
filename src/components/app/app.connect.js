import App from "@components/app/app";
import {connect} from "react-redux";
import {sortedOffers} from "../../store/selectors";
import {init} from "./actions/init";

const mapStateToProps = ({DATA, PROCESS, USER}) => ({
  city: PROCESS.city,
  cities: PROCESS.cities,
  reviews: DATA.reviews,
  offers: sortedOffers({DATA, PROCESS}),
  authorizationStatus: USER.authorizationStatus,
});

const mapDispatchToProps = (dispatch) => ({
  initialLoad: () => init(dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
