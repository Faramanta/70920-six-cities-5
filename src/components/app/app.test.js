import renderer from 'react-test-renderer';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import reducer from "@store/redusers/root-reducer";
import {offers} from "../../mocks/mocks";

import {App} from "./app";

const store = createStore(reducer);

jest.mock(`../map/map`, () => `Map`);

it(`App render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <App
            offers={offers}
            city={`Amsterdam`}
            hoverOfferCardId={1}
            authorizationStatus={`NO_AUTH`}
          />
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
