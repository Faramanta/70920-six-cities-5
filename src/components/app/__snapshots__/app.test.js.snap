// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render correctly 1`] = `
<div
  style={
    Object {
      "left": "50%",
      "position": "absolute",
      "top": "50%",
      "transform": "translate(-50%, -50%)",
    }
  }
>
  <p
    style={
      Object {
        "display": "block",
        "fontSize": "34px",
        "textAlign": "center",
      }
    }
  >
    Loading...
  </p>
</div>
`;
